pkgs/by-name/as/asfa/package.nix

```diff
@@ -0,0 +1,63 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  stdenv,
+  darwin,
+  help2man,
+  openssl,
+  pkg-config,
+}:
+let
+  version = "0.10.0";
+in
+rustPlatform.buildRustPackage {
+  pname = "asfa";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "obreitwi";
+    repo = "asfa";
+    rev = "v${version}";
+    hash = "sha256-MnhnwtZmPFhOuiqNiaxJnPu88JOdlpvyVy0YGphblBc=";
+  };
+
+  cargoHash = "sha256-/bRBP/NzcNOXl/nANeOYouUAo3NNbtbV9fxIJrNajYQ=";
+
+  buildInputs = [
+    openssl
+  ] ++ lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.SystemConfiguration ];
+
+  nativeBuildInputs = [
+    help2man
+    pkg-config
+  ];
+
+  doCheck = false;
+
+  postInstall = ''
+    mkdir -p "$out/man/man1"
+    help2man -o "$out/man/man1/asfa.1" "$out/bin/asfa"
```

Comment:
Since you are creating manpages, do these things:

1.  Modify outputs:

   ```nix
      outputs = [ "out" "man" ];
   ```

2. create the manpages at `postBuild` instead of `postInstall` - yes, do not create them directly at the final output dir;
3. install them at `postInstall` via `installManpage *.1` (installManpage comes from `installShellFiles` package).

Reply:
Thanks, adjusted.

Reply:
> create the manpages at postBuild instead of postInstall - yes, do not create them directly at the final output dir;

Turns out help2man hard-codes the name supplied binary so you can't replace `$out/bin/asfa` with `cargo run --release --` during the build phase. 

Found no easy & stable way of determining where the build products are stored (the cargo install hook seems to jump through [quite some hoops](https://github.com/NixOS/nixpkgs/blob/7fc3a682e8ae80681af91b84eea9a54583f21543/pkgs/build-support/rust/hooks/cargo-install-hook.sh#L17-L43) which I did not want to couple to).

Hence, the more straight-forward solution is to still create the the manpages in the `postInstall`-hook but then install them via `installManPage`. :+1: